### PR TITLE
Savestates/System.cpp: Realized fix for suspend mode

### PR DESCRIPTION
**Pull request by elad335**
Third time trying to implement it correctly, this time is because I have heard that the bug was only affecting game collections. This made me realize that savestate hiding was not done for all games when booting regularly.
Also move savestate hiding to after SPU/PPU cache were built which makes more sense.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -945,6 +918,8 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 				m_cat.clear();
 			}
 
+			m_path_old = m_path:
```
Comment (mzji) [inline rpcs3/Emu/System.cpp:921]: `:` -> `;` ?

Comment (elad335) [reply rpcs3/Emu/System.cpp:921]: Never! never surrender to semicolons!

File: rpcs3/Emu/System.cpp [hunk 2]
```diff
@@ -1984,6 +1959,32 @@ void Emulator::FixGuestTime()
...
 		CallFromMainThread([this]
 		{
+			// Mark a known savestate location and the one we try to boot (in case we boot a moved/copied savestate)
+			if (g_cfg.savestate.suspend_emu)
+			{
+				for (std::string old_path : std::initializer_list<std::string>{m_ar ? m_path_old : "", m_title_id.empty() ? "" : get_savestate_path(m_title_id, m_path_old)})
+				{
+					if (old_path.empty())
+					{
+					continue;
```
Comment (Megamouse) [inline rpcs3/Emu/System.cpp:1969]: Indentation

**Comment by Nekotekina** [comment]
Use squash when merging to create commit description

File: rpcs3/Emu/System.cpp (hunk 2 above)
Comment (elad335) [reply rpcs3/Emu/System.cpp]: Fixed.

File: rpcs3/Emu/System.cpp [hunk 3]
```diff
@@ -945,6 +918,8 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 				m_cat.clear();
 			}
 
+			m_path_old = m_path;
```
Comment (Megamouse) [inline rpcs3/Emu/System.cpp:921]: Why is this necessary?

**Comment by Triticum0** [comment]
This breaks the 1942: Joint Strike causing it to not make it past booting
